tensorflow/c/experimental/gradients/math_grad.cc

```diff
@@ -221,6 +221,36 @@ class NegGradientFunction : public GradientFunction {
   ~NegGradientFunction() override {}
 };
 
+class SubGradientFunction : public GradientFunction {
+ public:
+  Status Compute(Context* ctx, const IncomingGradients& grad_inputs,
+                 vector<AbstractTensorHandle*>* grad_outputs) override {
+    /* Given upstream grad U and a Sub op A-B, the gradients are:
+     *
+     *    dA =  U
+     *    dB = -U
+     *
+     */
+
+    grad_outputs->resize(2);
+
+    // Grad for A
+    DCHECK(grad_inputs[0]);
+    (*grad_outputs)[0] = grad_inputs[0];
```

Comment:
Since we are returning a reference to this tensor you need to call `grad_inputs[0]->Ref();` as well.

Reply:
Done